### ISSUE TITLE
fetch thread in KafkaSource exit abnormally and randomly  

### ISSUE THREAD

**Issue by huafengw** [gearpump/gearpump#1872]
Here is the related exception:
```
2016-01-15 14:22:34,652 ERROR redirect: Exception in thread "Thread-9" kafka.common.UnknownException
2016-01-15 14:22:34,653 ERROR redirect:         at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2016-01-15 14:22:34,653 ERROR redirect:         at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2016-01-15 14:22:34,653 ERROR redirect:         at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2016-01-15 14:22:34,653 ERROR redirect:         at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
2016-01-15 14:22:34,653 ERROR redirect:         at java.lang.Class.newInstance(Class.java:442)
2016-01-15 14:22:34,653 ERROR redirect:         at kafka.common.ErrorMapping$.exceptionFor(ErrorMapping.scala:86)
2016-01-15 14:22:34,653 ERROR redirect:         at io.gearpump.streaming.kafka.lib.consumer.KafkaConsumer$$anonfun$1.apply(KafkaConsumer.scala:46)
2016-01-15 14:22:34,653 ERROR redirect:         at io.gearpump.streaming.kafka.lib.consumer.KafkaConsumer$$anonfun$1.apply(KafkaConsumer.scala:38)
2016-01-15 14:22:34,653 ERROR redirect:         at io.gearpump.streaming.kafka.lib.consumer.KafkaConsumer.hasNextHelper$1(KafkaConsumer.scala:87)
2016-01-15 14:22:34,653 ERROR redirect:         at io.gearpump.streaming.kafka.lib.consumer.KafkaConsumer.hasNext(KafkaConsumer.scala:91)
2016-01-15 14:22:34,653 ERROR redirect:         at io.gearpump.streaming.kafka.lib.consumer.FetchThread$$anonfun$fetchMessage$1.apply(FetchThread.scala:91)
2016-01-15 14:22:34,653 ERROR redirect:         at io.gearpump.streaming.kafka.lib.consumer.FetchThread$$anonfun$fetchMessage$1.apply(FetchThread.scala:88)
2016-01-15 14:22:34,653 ERROR redirect:         at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:142)
```

and Kafka's error log:
```
[2016-01-15 14:22:34,647] ERROR [Replica Manager on Broker 1]: Error when processing fetch request for partition [ad-events,3] offset 5423844 from consumer with correlation id 0. Possible cause: Attempt to read with a maximum offset (5423761) less than the start offset (5423844). (kafka.server.ReplicaManager)
```

**Comment by manuzhang** [comment]
could be a kafka bug https://issues.apache.org/jira/browse/KAFKA-725

**Comment by guozhangwang** [comment]
@manuzhang Yes this indeed seems relevant, will review and merge in the patch.

**Comment by manuzhang** [comment]
[KAFKA-725](https://issues.apache.org/jira/browse/KAFKA-725) has been resolved. Please open a jira at https://issues.apache.org/jira/browse/GEARPUMP if this happens again.

**Comment by huafengw** [comment]
Should we update the Kafka version?

**Comment by manuzhang** [comment]
This is fixed in kafka 0.10 so I think it will take a while.
